### PR TITLE
Add simd_neg platform intrinsic

### PR DESCRIPTION
**Pull request by SparrowLii**
Stdarch needs to add simd_neg to support the implementation of vneg neon instructions. Look [here](https://github.com/rust-lang/stdarch/pull/1087)

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/intrinsic.rs [hunk 1]
```diff
@@ -1628,7 +1628,7 @@ unsupported {} from `{}` with element `{}` of size `{}` to `{}`"#,
             out_elem
         );
     }
-    macro_rules! arith {
+    macro_rules! arith_bin {
```
Comment (joshtriplett) [inline compiler/rustc_codegen_llvm/src/intrinsic.rs:1631]: These just get defined once and invoked once, so they don't need to be abbreviated. How about `arith_binary` and `arith_unary`?

**Comment by joshtriplett** [comment]
Naming nit aside, LGTM.

File: compiler/rustc_codegen_llvm/src/intrinsic.rs (hunk 1 above)
Comment (SparrowLii) [reply compiler/rustc_codegen_llvm/src/intrinsic.rs]: Thanks for pointing. It has been corrected.

**Pull request merged**